cpp/src/umap/knn_graph/algo.cuh

```diff
@@ -57,25 +62,45 @@ inline void launcher(const raft::handle_t& handle,
                      const ML::UMAPParams* params,
                      cudaStream_t stream)
 {
-  std::vector<float*> ptrs(1);
-  std::vector<int> sizes(1);
-  ptrs[0]  = inputsA.X;
-  sizes[0] = inputsA.n;
-
-  raft::spatial::knn::brute_force_knn(handle,
-                                      ptrs,
-                                      sizes,
-                                      inputsA.d,
-                                      inputsB.X,
-                                      inputsB.n,
-                                      out.knn_indices,
-                                      out.knn_dists,
-                                      n_neighbors,
-                                      true,
-                                      true,
-                                      static_cast<std::vector<int64_t>*>(nullptr),
-                                      params->metric,
-                                      params->p);
+  if (params->build_algo == ML::UMAPParams::graph_build_algo::BRUTE_FORCE_KNN) {
+    std::vector<float*> ptrs(1);
+    std::vector<int> sizes(1);
+    ptrs[0]  = inputsA.X;
+    sizes[0] = inputsA.n;
+
+    raft::spatial::knn::brute_force_knn(handle,
+                                        ptrs,
+                                        sizes,
+                                        inputsA.d,
+                                        inputsB.X,
+                                        inputsB.n,
+                                        out.knn_indices,
+                                        out.knn_dists,
+                                        n_neighbors,
+                                        true,
+                                        true,
+                                        static_cast<std::vector<int64_t>*>(nullptr),
+                                        params->metric,
+                                        params->p);
+  } else {  // nn_descent
+    // n_neighbors should be smaller than the graph degree computed by nn descent
+    assert(n_neighbors <= params->nn_descent_params.graph_degree);
```

Comment:
Actually when I think of it, since it's throwing an exception, it's probably better to just add a message to the `assert` function. Usually, we use the `ASSERT` macro from gtest I believe. Or simply throwing a C++ exception. The end goal is to help users know why the process is crashing with an informative message.

Reply:
Let's use RAFT_EXPECTS instead of ASSERT please.